Azure/azure-cli
# [RDBMS] `az mysql flexible-server create/update/restore/geo-restore/replica`: Add BYOK for MySQL Flexible Server

**Related commands**
`az mysql flexible-server create --key key --identity identity`
`az mysql flexible-server update --key key --identity identity`
`az mysql flexible-server restore`
`az mysql flexible-server geo-restore`
`az mysql flexible-server replica`

**Description**
This PR enables data encryption for MySQL Flexible Server with customer-managed identity and customer-managed key.

- Add arguments `--key` and `--identity` for create and update server.
- Add argument `--disable-data-encryption` for update server.
- Send key and identity properties for replica create, restore from a backup and geo-restore.
- Add arguments `--backup-key` and `--backup-identity` for all of the above, for geo-redundant servers.

**Testing Guide**
See [_helptext_mysql.py](https://github.com/Azure/azure-cli/pull/23197/files#diff-e7e0e994269ad38b069f8618e75c536cf5a7ae0fe79514277bbb332f00cee7afR96) and [test_rdbms_flexible_commands.py](https://github.com/Azure/azure-cli/pull/23197/files#diff-64267f415c845d242b32853a748c529022a3f513732312caec287632cf5a26b3R509)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
